opc/resource_lbaas_load_balancer.go

```diff
@@ -0,0 +1,327 @@
+	}
+
+	if enabled, ok := d.GetOk("enabled"); ok {
+		input.Disabled = getDisabledStateKeyword(enabled.(bool))
+	}
+
+	if ipNetwork, ok := d.GetOk("ip_network"); ok {
+		input.IPNetworkName = ipNetwork.(string)
+	}
+
+	if scheme, ok := d.GetOk("scheme"); ok {
+		input.Scheme = lbaas.LoadBalancerScheme(scheme.(string))
+	}
+
+	if serverPool, ok := d.GetOk("server_pool"); ok {
+		input.OriginServerPool = serverPool.(string)
+	}
+
+	if parent, ok := d.GetOk("parent_load_balancer"); ok {
+		input.ParentLoadBalancer = parent.(string)
+	}
+
+	permittedClients := getStringSet(d, "permitted_clients")
+	if len(permittedClients) != 0 {
+		input.PermittedClients = permittedClients
+	}
+
+	permittedMethods := getStringSet(d, "permitted_methods")
+	if len(permittedMethods) != 0 {
+		input.PermittedMethods = permittedMethods
+	}
+
+	policies := getStringSet(d, "policies")
+	if len(policies) != 0 {
+		input.Policies = policies
+	}
+
+	tags := getStringSet(d, "tags")
+	if len(tags) != 0 {
+		input.Tags = tags
+	}
+
+	info, err := lbClient.CreateLoadBalancer(&input)
+	if err != nil {
+		return fmt.Errorf("Error creating Load Balancer: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", info.Region, info.Name))
+	return resourceOPCLoadBalancerRead(d, meta)
+}
+
+func resourceOPCLoadBalancerRead(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	lbClient := lbaasClient.LoadBalancerClient()
+
+	s := strings.Split(d.Id(), "/")
+	lb := lbaas.LoadBalancerContext{
+		Region: s[0],
+		Name:   s[1],
+	}
+
+	result, err := lbClient.GetLoadBalancer(lb)
+	if err != nil {
+		// LoadBalancer does not exist
+		if client.WasNotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Load Balancer %s: %s", d.Id(), err)
+	}
+
+	if result == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("canonical_host_name", result.CanonicalHostName)
+	d.Set("cloudgate_capable", result.CloudgateCapable)
+	d.Set("enabled", getEnabledState(result.Disabled))
+	d.Set("description", result.Description)
+	d.Set("ip_network", result.IPNetworkName)
+	d.Set("name", result.Name)
+	d.Set("parent_load_balancer", result.ParentLoadBalancer)
+	d.Set("region", result.Region)
+	d.Set("scheme", result.Scheme)
+	d.Set("server_pool", result.OriginServerPool)
+	d.Set("uri", result.URI)
+
+	if err := setStringList(d, "balancer_vips", result.BalancerVIPs); err != nil {
+		return err
+	}
+	if err := setStringList(d, "permitted_clients", result.PermittedClients); err != nil {
+		return err
+	}
+	if err := setStringList(d, "permitted_methods", result.PermittedMethods); err != nil {
+		return err
+	}
+	if err := setStringList(d, "policies", result.Policies); err != nil {
+		return err
+	}
+	if err := setStringList(d, "tags", result.Tags); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceOPCLoadBalancerUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Let's also update a LoadBalancer in a test as noted above

Reply:
done